src/Chullo.php

```diff
@@ -357,28 +296,23 @@ protected function prepareUri($uri, $transaction = "") {
      * @param array     $destination    Destination URI
      * @param string    $transaction    Transaction id
      *
-     * @return string
+     * @return string   Uri of newly created copy or null if failed
      */
-    public function copyResource($uri,
-                                 $destination,
-                                 $transaction = "") {
-        // Ensure uri takes transaction into account.
-        $uri = $this->prepareUri($uri, $transaction);
-        // Create destinsation URI
-        $destination_uri = $this->prepareUri($destination, $transaction);
-        // Create destination array
-        $options = [
-          'headers' => [
-            'Destination' => $destination_uri,
-            'Overwrite'   => 'T'
-            ],
-        ];
-        $response = $this->client->request(
-          'COPY',
-          $uri,
-          $options
+    public function copyResource(
+        $uri,
+        $destination,
+        $transaction = ""
+    ) {
+        $response = $this->api->copyResource(
+            $uri,
+            $destination,
+            $transaction
         );
 
+        if ($response->getStatusCode() != 201) {
+            return null;
+        }
+
         // Return the value of the location header
```

Comment:
We should consider making this a utility function.  It happens a few times.  Guess it deserves its own function.